infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/SwitchLocationTrackLink.kt

```diff
@@ -97,16 +103,19 @@ fun getDuplicateMatches(
         emptyList()
     } else if (matchRanges.isEmpty() && duplicateOf == mainTrackId) {
         // Marked as duplicate, but no matches -> something is likely wrong
-        listOf(-1 to DuplicateStatus(DuplicateMatch.NONE, duplicateOf, null, null))
+        listOf(-1 to DuplicateStatus(DuplicateMatch.NONE, duplicateOf, null, null, null))
     } else {
         matchRanges.map { range ->
             val match = if (range == 0..duplicateTrackSplitPoints.lastIndex) DuplicateMatch.FULL
             else DuplicateMatch.PARTIAL
+            val start = duplicateTrackSplitPoints[range.first]
+            val end = duplicateTrackSplitPoints[range.last]
```

Comment:
Lienee ok, mutta tällaiset ovat hieman kyllä riskialttiita index-out-of-bounds -mahdollisuuksien vuoksi (tarkempi menettely olisi vielä tarkistaa, että kyseiset indeksit ovat olemassa ennen kuin niistä yrittää hakea tietoa, hanskaten virhetilanteet järkevästi).